lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb

```diff
@@ -137,32 +137,54 @@ def embedded_content_settings
         #         the pod targets.
         #
         def settings_to_import_pod_targets
-          if target.requires_frameworks?
-            build_pod_targets = pod_targets.select(&:should_build?)
-            framework_header_search_paths = build_pod_targets.map do |target|
-              "#{target.build_product_path}/Headers"
+          header_search_paths = []
+          silenced_header_search_paths = []
+          silenced_frameworks = []
+          pod_targets.each do |target|
+            target_header_search_paths = if target.should_build? && target.requires_frameworks?
+                                           ["#{target.build_product_path}/Headers"]
+                                         else
+                                           target.sandbox.public_headers.search_paths(target.platform)
+                                         end
+            header_search_paths += target_header_search_paths
+            if target.inhibit_warnings?
+              silenced_header_search_paths += target_header_search_paths
+              if target.requires_frameworks? && target.should_build?
+                silenced_frameworks.push(target.configuration_build_dir)
+              end
+            end
+          end
+
+          header_search_paths.uniq!
+          silenced_header_search_paths.uniq!
+
+          build_settings = {
+            # Make headers discoverable with any syntax (quotes, brackets,
+            # @import, etc.)
+            'HEADER_SEARCH_PATHS' => XCConfigHelper.quote(header_search_paths),
+          }
+
+          # Use -isystem and -iframework for headers we want to treat as
```

Comment:
what is `-iframework` for? it doesnt seem to show up in the integration specs

Reply:
`-iframework` is used in the same way `-isystem` is.

Note that despite the name, `-iframework` has noting to do with `-framework`:
From https://gcc.gnu.org/onlinedocs/gcc/Darwin-Options.html :

> **-iframework _dir_**
> Like -F except the directory is a treated as a system directory. The main difference between this -iframework and -F is that with -iframework the compiler does not warn about constructs contained within header files found via dir.

(`-F` being an equivalent of `FRAMEWORK_SEARCH_PATHS`)

The behavior I implemented in this PR is as follow :

- We use `HEADER_SEARCH_PATHS` to search for "regular" headers and `FRAMEWORK_SEARCH_PATHS` to find framework headers (as before)
- We only add the headers / framework path in `OTHER_C_FLAGS` using the `-isystem` or `-iframework` flags if the pod's warnings are silenced.

I should probably add some integration tests for all that. I'll try to do this asap but I don't have much time to work on this at the moment.

-------------------

Sidenote: I saw you added some comment to make `#import <...>` the only valid option in CocoaPods 2.0. I think it'll be difficult to get this behavior without silencing the headers (because that's how `-isystem` works). You'll end up in the invert situation of the one we're in today : pods without silenced warning won't report those warnings in header files (but I may be mistaken, I'm no expert in compiler flags 🙃 ).

Reply:
> Sidenote: I saw you added some comment to make #import <...> the only valid option in CocoaPods 2.0. I think it'll be difficult to get this behavior without silencing the headers (because that's how -isystem works). You'll end up in the invert situation of the one we're in today : pods without silenced warning won't report those warnings in header files (but I may be mistaken, I'm no expert in compiler flags 🙃 ).

I think the (eventual) goal is to make everything either static or dynamic frameworks :)